### PR TITLE
Made sampling based on UserId configurable

### PR DESCRIPTION
**Pull request by d3r3kk**
- Moved the configuration out as far as the AdaptiveSamplingTelemetryProcessor
- Added unit test to ensure sampling with/without user id is tested
- Updated changelog for beta 2.5.0-beta2
- Updated PublicAPI.Unshipped.txt

Linked to issue #625

For significant contributions please make sure you have completed the following items:

- [x] Design discussion issue #625 
- [ ] Changes in public surface reviewed
- [x] CHANGELOG.md updated

### CODE REVIEW

File: src/ServerTelemetryChannel/Managed/Shared/SamplingTelemetryProcessor.cs [hunk 1]
```diff
@@ -125,6 +126,17 @@ public string IncludedTypes
...
         /// <summary>
+        /// Gets or sets a value indicating whether or not we should calculate sampling based on the Context.User.Id field.
+        /// Default is false.
+        /// 
+        /// <remarks>
+        /// Calculation of the sampling score based on user_Id field is disabled by default to ensure successful correlation across many 
+        /// services on operation_Id field.
+        /// </remarks>
+        /// </summary>
+        public bool AllowSamplingBasedOnUserId { get; set; }
```
Comment (SergeyKanzhelev) [inline src/ServerTelemetryChannel/Managed/Shared/SamplingTelemetryProcessor.cs]: `Allow` is not the best verb here. Maybe `Calculate` like in `CalculateSamplingScoreFromUserId` or `Use` like in `UseUserIdSamplingScore`

File: src/ServerTelemetryChannel/Managed/Shared/SamplingTelemetryProcessor.cs [hunk 2]
```diff
@@ -125,6 +126,17 @@ public string IncludedTypes
         }
 
         /// <summary>
+        /// Gets or sets a value indicating whether or not we should calculate sampling based on the Context.User.Id field.
+        /// Default is false.
+        /// 
+        /// <remarks>
+        /// Calculation of the sampling score based on user_Id field is disabled by default to ensure successful correlation across many
```
Comment (SergeyKanzhelev) [inline src/ServerTelemetryChannel/Managed/Shared/SamplingTelemetryProcessor.cs]: nit: why are you switching to Kusto names of the fields in remark?

File: src/ServerTelemetryChannel/Managed/Shared/Implementation/SamplingScoreGenerator.cs [hunk 3]
```diff
@@ -13,12 +13,13 @@ internal static class SamplingScoreGenerator
         /// Generates telemetry sampling score between 0 and 100.
         /// </summary>
         /// <param name="telemetry">Telemetry item to score.</param>
+        /// <param name="allowUserId">Sampling score can be determined by using the Context.User.Id information, default is false.</param>
```
Comment (SergeyKanzhelev) [inline src/ServerTelemetryChannel/Managed/Shared/Implementation/SamplingScoreGenerator.cs]: Add remark that is can be used for backward compatibility

File: Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/SamplingScoreGeneratorTest.cs [hunk 4]
```diff
@@ -27,8 +27,8 @@ public void SamplingScoreGeneratedUsingUserIdIfPresent()
             requestTelemetry.Context.User.Id = userId;
             requestTelemetry.Context.Operation.Id = GenerateRandomOperaitonId();
 
-            var eventTelemetrySamplingScore = SamplingScoreGenerator.GetSamplingScore(eventTelemetry);
-            var requestTelemetrySamplingScore = SamplingScoreGenerator.GetSamplingScore(requestTelemetry);
+            var eventTelemetrySamplingScore = SamplingScoreGenerator.GetSamplingScore(eventTelemetry, true);
```
Comment (SergeyKanzhelev) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/SamplingScoreGeneratorTest.cs]: update the name of the test method. `IfPresentAndConfigured`

File: Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/SamplingScoreGeneratorTest.cs [hunk 5]
```diff
@@ -44,20 +44,46 @@ public void SamplingScoreGeneratedUsingOperationIdIfPresent()
...
+
+            var eventTelemetry = new EventTelemetry();
+            eventTelemetry.Context.Operation.Id = operationId;
+            eventTelemetry.Context.User.Id = userId;
+
+            var requestTelemetry = new RequestTelemetry();
+            requestTelemetry.Context.Operation.Id = operationId;
+            requestTelemetry.Context.User.Id = userId;
+
+            var eventTelemetrySamplingScore = SamplingScoreGenerator.GetSamplingScore(eventTelemetry, true);
```
Comment (SergeyKanzhelev) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/SamplingScoreGeneratorTest.cs]: ever test it with `false`?

**Comment by d3r3kk** [comment]
Shutting this PR down, we will simply remove our calculation of sampling score based on Context.User.Id altogether.

**Pull request closed without merging**